frontend/src/views/HomeView.vue

```diff
@@ -1,9 +1,111 @@
 <script setup lang="ts">
-import ApiTest from "@/components/ApiTest.vue";
+import { ref } from 'vue';
+import WeatherView from "@/components/WeatherView.vue";
+import Modal from "@/components/Modal.vue";
+
+const isModalOpen = ref(false);
+const userSelected = ref<any>(null);
+
+const openWeatherModal = (user: any) => {
+  userSelected.value = user;
+  isModalOpen.value = true;
+};
+
+const closeModal = () => {
+  isModalOpen.value = false;
+  userSelected.value = null;
+};
+
+const formatTime = (dateTime: string) => {
```

Comment:
I like that you defined this in the setup rather than trying to cram it into the view logic. Also good use of native js functionality rather than pulling in an unnecessary dependency 👍🏻